Hints.py

```diff
@@ -359,10 +362,17 @@ def get_good_item_hint(spoiler, world, checked):
 
 def get_specific_item_hint(spoiler, world, checked):
     itemname = world.item_hints.pop(0)
-    locations = [location for location in world.get_filled_locations()
-            if is_not_checked(location, checked) and \
-            location.item.name == itemname and \
-            not location.locked]
+    if itemname=="Bottle":
+        locations = [location for location in world.get_filled_locations()
+        if is_not_checked(location, checked) and \
+        location.item.name in location.item.name in ("Bottle", "Bottle with Red Potion","Bottle with Green Potion", "Bottle with Blue Potion", 
+                                                     "Bottle with Fairy", "Bottle with Fish", "Bottle with Blue Fire", "Bottle with Bugs", "Bottle with Big Poe", "Bottle with Poe") and \
```

Comment:
The main idea for including "Bottle with Big Poe" here is that it is easy to empty out that bottle without doing any logic checks (whereas giving hint for Ruto's Letter would also mean ensuring access to Domain). The only situations where that might not be the case are in the more complex versions of ER and if DoT is closed. In both cases the bottle logic is going to be the least of your problems. 
I'm not familiar with all the details of the filling algorithm, is it possible to have all three non-Ruto's Letter bottles be Big Poe bottles? That was my real concern with not including it, since if that situation turned up the hint system would throw an error.

In terms of the second part, do you know what qualifies as a "real" bottle for logic? I can include that as a separate list and check if bingo hints are being used

Reply:
The items that count as bottles are in ItemList.py marked as 'bottle' and `ItemInfo.isBottle(name)` is what you'd use to look up whether something is a bottle. The gotchas are that `Rutos Letter` and `Bottle with Big Poe` are not bottles, while `Deliver Letter` and `Sell Big Poe` (which you get from turning in the non-bottle bottles) are bottles. We do that to account for ability to empty the bottles in logic.

My question was rather about whether you'd count them toward a goal like "get 3 bottles" and not have players concerned about whether the bottles can be emptied.

Reply:
I tested out putting "Sell Big Poe" and "Deliver Letter" in the get_specific_item_hint to account for the logic needed to empty the bottles, however that throws an error. Not surprising, since that function is designed to give the location for the item and it isn't clear what the location of "Deliver Letter" is. I did add a separate list for non-bingo hint distributions that includes Ruto's Letter, since for "normal" rando you're ensured to be able to empty it if needed. 

There aren't any goals in the style of "get 3 bottles". Instead the goals are of the form "At least X bottle slots filled", so having a big poe in them is not a concern for those goals (and are completed regardless of bottle contents). The idea for adding the bottle hints was more for goals such as "All 9 Soft Soil Skulltulas" that specifically require a fillable bottle.  In those cases it's not guaranteed that the player has access to Zora's Domain as child to deliver the letter. In ER it's also not guaranteed that you can sell the big poe for that purpose, but it is possible to end up with big poe bottles as the only non-Ruto's Letter bottles. If big poe bottles were excluded from the list and that happened, the generator would throw an error. In my view, it was better to have a less useful hint (pointing to a bottle that you are very likely but not guaranteed to be able to empty) rather than risking the generator failing.

Reply:
Yeah, they're pseudo-items, always in the same place. I think leaving the hints pointing at the real bottles no matter their contents sounds fine. Your "bottle slot" goals aren't assuming you can use the bottle as a bottle, is what I mean.

I am not so certain about this change for Ruto's Letter in the non-bingo hint dist. It should probably exclude letter and big poe if we're not doing bingo. Honestly, now that I think about it, I don't know why this should change non-bingo hint distributions; the previous code only hinted the exact item "Bottle" for a specific hint "Bottle".

Reply:
The bottle slot goals aren't assuming the use of bottle as a bottle, no. But the goals like the soft soils ones do assume you can use it as a bottle. Because of the way the goals are set up, you won't get bottle hints for the bottle slot goals (since those types of goals are focused on finding the items, so hints to the location would defeat the point). 

That's a fair point for non-hint distribution. If you really want *any* bottle, you can wait for mracsys' changes and use #Bottle LUL I'll change it back to be empty bottle specific

Reply:
Ok, resolving this comment now that I understand and the current agreement is: when bingo needs to hint a bottle, this will hint any bottle object other than Ruto's and assume one can reach Sell Big Poe in case it picks a Big Poe. Thanks!